### DISCUSSION TITLE
The value is empty during the second query.

### DISCUSSION THREAD

**Discussion by gaoyuan98** [milvus-io/milvus#25969] [Q&A and General discussion] [answered]
这个是官方的例子，第一次执行时，query可以查询出数据，相同的代码 在调用第二次时就会返回[]，如下图。我现在有点搞不清原因是什么。

![image](https://github.com/milvus-io/milvus/assets/44916277/ff0205c0-b605-4b0f-ab2b-9ed87f30ae2f)

这时，我不插入数据，直接加载这个表查询，也是为空，希望可以解答一下
![image](https://github.com/milvus-io/milvus/assets/44916277/325b3a6e-b3ca-4ea1-92ed-68e5f5adf69b)

**Comment by yhmo** [comment, marked as answer]
你把那个expr="film_id <= 1" 改成 expr="film_id <= 10"看一下结果
然后再把query()里的limit=1改成limit=10，再观察一下结果
然后再把query()里的offset=1去掉，再观察一下结果

expr决定了过滤出部分向量进行ANN查询。expr="film_id <= 1"，符合这个过滤条件的应该只有两个，分别是film_id=0和film_id=1
offset是指从得到的结果集里往后面偏移n个位置取结果
limit是指从某个位置开始取m个结果。如果offset=1, limit=1，就是从1号位置取1个结果。如果去掉offset=1,就是从0号位置取1个结果

**Comment by yhmo** [comment]
另外：milvus社区里鼓励使用英文，尽量使用英文提问和对话，谢谢

**Comment by gaoyuan98** [comment]
query method drop offset=1 success 
![image](https://github.com/milvus-io/milvus/assets/44916277/2257ec6b-05fe-4c5e-bf7f-1123d734a5c8)
